src/zenml/cli/base.py

```diff
@@ -63,19 +66,116 @@ def init(path: Optional[Path]) -> None:
     )
 
 
-@cli.command("clean")
-@click.option("--yes", "-y", type=click.BOOL, default=False)
-def clean(yes: bool = False) -> None:
-    """Clean everything in repository.
+def _delete_local_artifact_metadata(force_delete: bool = False) -> None:
+    """Delete local metadata and artifact stores from the active stack.
 
     Args:
-      yes: bool:  (Default value = False)
+      force_delete: Whether to force delete the metadata and artifact stores."""
+    if not force_delete:
+        confirm = confirmation(
+            "DANGER: This will completely delete anything inside the folders for the following stack components: \n"
+            "- local metadata store \n"
+            "- local artifact store. \n\n"
+            "Are you sure you want to proceed?"
+        )
+        if not confirm:
+            declare("Aborting clean.")
+            return
+
+    repo = Repository()
+    if repo.active_stack:
+        metadata_store_path = (
```

Comment:
As any stack component now has an optional `local_path` property, is this something we could maybe extend to not only handle the artifact/metadata store here?

Reply:
I don't understand this comment. Are you suggesting that I could leverage the `local_path` property as a way of discovering whether a component was local or not?

What does 'not only handle' mean?

Reply:
Sorry, could have been clearer on that. Any `StackComponent` can specify a `local_path` that is uses to store persistent information on disk. If we call `zenml clean --local` it would currently delete the content of the metadata and artifact store right? We could extend the functionality to also clean the local directories of other stack components, for example a feature store or container registry.

Reply:
Maybe also good to loop in @stefannica here to see if he thinks this makes sense or what was intended with the `--local` option

Reply:
I agree with @schustmi, you can loop through all the active stack components and, if you find `local_path` set to a non-empty value, delete that path (file or directory(.